internal/hcsoci/hcsdoc_wcow.go

```diff
@@ -61,57 +63,83 @@ func createWindowsContainerDocument(coi *createOptionsInternal) (interface{}, er
 		v2Container.GuestOs = &hcsschema.GuestOs{HostName: coi.Spec.Hostname}
 	}
 
-	if coi.Spec.Windows.Resources != nil {
-		if coi.Spec.Windows.Resources.CPU != nil {
-			if coi.Spec.Windows.Resources.CPU.Count != nil ||
-				coi.Spec.Windows.Resources.CPU.Shares != nil ||
-				coi.Spec.Windows.Resources.CPU.Maximum != nil {
-				v2Container.Processor = &hcsschema.Processor{}
-			}
-			if coi.Spec.Windows.Resources.CPU.Count != nil {
-				var hostCPUCount uint64
-				if coi.HostingSystem != nil {
-					// Normalize to UVM size
-					hostCPUCount = uint64(coi.HostingSystem.ProcessorCount())
-				} else {
-					hostCPUCount = uint64(runtime.NumCPU())
-				}
-				cpuCount := *coi.Spec.Windows.Resources.CPU.Count
-				if cpuCount > hostCPUCount {
-					logrus.Warnf("Changing requested CPUCount of %d to current number of processors, %d", cpuCount, hostCPUCount)
-					cpuCount = hostCPUCount
-				}
-				v1.ProcessorCount = uint32(cpuCount)
-				v2Container.Processor.Count = int32(cpuCount)
+	// CPU Resources
+	cpuNumSet := 0
+	cpuCount := oci.ParseAnnotationsCPUCount(coi.Spec, oci.AnnotationContainerProcessorCount, 0)
+	if cpuCount > 0 {
+		cpuNumSet++
+	}
+
+	cpuLimit := oci.ParseAnnotationsCPULimit(coi.Spec, oci.AnnotationContainerProcessorLimit, 0)
+	if cpuLimit > 0 {
+		cpuNumSet++
+	}
+
+	cpuWeight := oci.ParseAnnotationsCPUWeight(coi.Spec, oci.AnnotationContainerProcessorWeight, 0)
+	if cpuWeight > 0 {
+		cpuNumSet++
+	}
+
+	if cpuNumSet > 1 {
+		return nil, fmt.Errorf("invalid spec - Windows Process Container CPU Count: '%d', Limit: '%d', and Weight: '%d' are mutually exclusive", cpuCount, cpuLimit, cpuWeight)
+	} else if cpuNumSet == 1 {
+		var hostCPUCount int32
+		if coi.HostingSystem != nil {
+			// Normalize to UVM size
+			hostCPUCount = coi.HostingSystem.ProcessorCount()
+		} else {
+			hostCPUCount = int32(runtime.NumCPU())
+		}
+		if cpuCount > hostCPUCount {
+			log := logrus.WithField(logfields.ContainerID, coi.ID)
+			if coi.HostingSystem != nil {
+				log.Data[logfields.UVMID] = coi.HostingSystem.ID()
 			}
-			if coi.Spec.Windows.Resources.CPU.Shares != nil {
-				v1.ProcessorWeight = uint64(*coi.Spec.Windows.Resources.CPU.Shares)
-				v2Container.Processor.Weight = int32(v1.ProcessorWeight)
+			log.Warningf("Changing user requested CPUCount: %d to current number of processors: %d", cpuCount, hostCPUCount)
+			cpuCount = hostCPUCount
+		}
+
+		v1.ProcessorCount = uint32(cpuCount)
+		v1.ProcessorMaximum = int64(cpuLimit)
+		v1.ProcessorWeight = uint64(cpuWeight)
+
+		if cpuCount == 0 {
+			// TODO: JTERRY75 - There is a Windows platform bug for V2 that we
```

Comment:
Might be worth adding the VSO reference here

Reply:
Done.